key-concepts/architecture.md

```diff
@@ -28,30 +25,41 @@ The GraphQL Server is a separate web server providing a GraphQL API (more on the
 
 ### Squid
 
-A Squid is a query node for ingesting, transforming, and presenting Substrate blockchain data.
+A Squid is a query node for transforming, and presenting Substrate blockchain data. It consists of:
+
+* **Processor**: each node has one but can have multiple if connected to multiple chains
+* **Database**: A PostgreSQL database where processed data is stored
+* **GraphQL Server**: every query node comes with a gateway to present processed data
 
 A Squid replaces direct gRPC node access with more performant API calls to Squid Archive gateways, allowing bandwidth reduction and quick synchronization of the API with the historical on-chain data. It can be run locally, in a server, or deployed to the Cloud, thanks to our SaaS solution.
 
 Thanks to the Subsquid framework, it is possible to:
 
-* define the [database schema](../recipes/how-to-define-a-database-schema.md), [data type, and entity definitions](../tutorial/generate-typescript-definitions.md)
+* define the [database schema](../recipes/define-a-squid-schema.md), [data type, and entity definitions](../recipes/generate-typescript-definitions.md)
 * transform and store chain data efficiently
 * present it thanks to the included [GraphQL server](broken-reference)
 
-Subsquid provides developers with a high-level GraphQL-like schema and codegen tools to model blockchain data with Entities and define data types.\
-The next step, which is the main customization needed, is to write code to describe the Event handlers, which map Events and Entities, telling the Processor how the blockchain Events affect the schema Entities.
+Subsquid provides developers with a high-level GraphQL-like schema and codegen tools to model blockchain data with Entities and define data types.
```

Comment:
That'a a bit vague. I think we need to emphasise type safety as this is main advantage of having typegen. Type safety is cool as it a) removes a unboxing/decoding boilerplate code b) eliminates the biggest pain point of incorrect data types due to wrong decoding, missing parameters or runtime upgrade. This can't be empathised enough as this is probable _the biggest_ pain point of developing mappings (e.g. with Subquery) -- one has to handle some unstructured data from events, and the data format may change block over block due to runtime upgrades! Without type safety and automation by typegen it's a total mess to handle all these changes.

Reply:
That's great. The last couple of sentences put it out really nicely. I'll incorporate your suggestion.